tests/org.eclipse.swt.tests/JUnit Tests/org/eclipse/swt/tests/junit/Test_org_eclipse_swt_browser_Browser.java

```diff
@@ -1238,18 +1250,9 @@ private void validateTitleChanged(String expectedTitle, Runnable browserSetFunc)
 	browserSetFunc.run();
 	shell.open();
 
-	boolean passed = waitForPassCondition(() -> actualTitle.get().equals(expectedTitle));
-	String errMsg = "";
-	if (!passed) {
-		if (actualTitle.get().length() == 0) {
-			errMsg = "Test timed out. TitleListener not fired";
-		} else {
-			errMsg = "\nExpected title and actual title do not match."
-					+ "\nExpected: " + expectedTitle
-					+ "\nActual: " + actualTitle;
-		}
-	}
-	assertTrue(errMsg + testLog.toString(), passed);
+	waitForPassCondition(() -> actualTitle.get().equals(expectedTitle));
+	assertTrue("Test timed out. TitleListener not fired", actualTitle.get().length() != 0);
```

Comment:
Would be good to log the testLog in case this assertion fails as well (like in the subsequent assertion).

Reply:
There would not be anything in the testLog in this, since the callback would have never executed. Do you think I should still add it?

Reply:
I see, then I think it's fine to leave it out. Would only be relevant if for some reasons the listener was fired but still the title is empty (or became empty afterwards again), but I do not see how that could happen.